### ISSUE TITLE
Wrong countKeys causes key collision of elements

### ISSUE THREAD

**Issue by knyipab** [ImpulseAdventure/GUIslice-Builder#54]
I'm not sure how I got to this bug. But after a long series of edit actions, now I have two elements with the key. Here's part of my project: 

[gui2.zip](https://github.com/ImpulseAdventure/GUIslice-Builder/files/4013874/gui2.zip)

From what I see in EnumFactory.java, I could spot a possible cause of the bug in `public void resetCounts(List<PagePane> pages)` from line 335 to 339: 
```
      for (Widget w : widgets) {
        sType = w.getType();
        sCount = CodeUtils.getKeyCount(w.getKey());
        resetCount(sType, c);
      }
```
It seems that the loop misses a line `c = Integer.valueOf(sCount);`. Adding the line should fix the problem. 

In `public void resetCount(String type, int c)`, 
```
    if (countKeys[idx] < c)
      countKeys[idx] = c;
```
I can see that countKeys can only get larger and larger except calling the `restore()` function so I think the root cause may be `restore()` miscalled somewhere else.

**Comment by knyipab** [comment]
Oh, I just forgot to mention, the key collision in my project happends to E_ELEM_TEXT16. There're two elements with E_ELEM_TEXT16 and key Text$16.

**Comment by Pconti31** [comment]
Assigned bug 130 will be in 0.13.b011
I think you are correct about the missing line `c = Integer.valueOf(sCount);`
Paul--

**Comment by Pconti31** [comment]
fixed in release 0.13.b011
